crates/uv/src/commands/pip/compile.rs

```diff
@@ -289,7 +289,7 @@ pub(crate) async fn pip_compile(
     );
 
     // Add all authenticated sources to the cache.
-    for index in index_locations.allowed_indexes() {
+    for index in index_locations.allowed_indexes().rev() {
         if let Some(credentials) = index.credentials() {
             uv_auth::store_credentials(index.raw_url(), credentials);
         }
```

Comment:
Maybe we should have some sort of `uv_auth::store_credentials` method that takes `index_locations` directly so we can avoid repeating this pattern everywhere and document why it's reversed and such?

Reply:
I'd prefer not to couple them for now. `uv_auth` has no dependencies as-is.